src/Invariants/domination/domination_numbers.jl

```diff
@@ -26,15 +26,15 @@ julia> compute(DominationNumber, g)
 """
 function compute(
     ::Type{DominationNumber},
-    g::AbstractGraph{T};
+    g::SimpleGraph{T};
```

Comment:
@joshyaffee Julia is really like functional programming language since it does not support object oriented programming (which is vastly different than Python), and as such, Julia relies on `AbstractTypes` and `Structs` to tell functions what to do. In the old code, the graph `g` was assumed to be of type `AbstractGraph`, which is the most general type of graph (these types come from `Graphs.jl`). 

What these means is that this function could have theoretically accepted any `AbstractGraph` type without reporting an error, such `SimpleGraph`, `DiGraph`, or `SimpleWeightedGraphs` types. However, this (and most of the functions in Invariants) only really make sense on `SimpleGraph` types. In fact, if a user passed in a `DiGraph` type, then this function would make an incorrect result without reporting an error. So most functions within the Invariants submodule need to only accept `SimpleGraph` types. 

Also, since `SimpleGraph` types are less general than the `AbstractGraph` type, the Julia compiler might be able to speed up because of this change.

Reply:
@RandyRDavila I'd just like to clarify: are Simple Graph objects necesarily undirected? I have seen some discrepancy about the definition, where some definitions are willing to classify diGraphs as simple as long as there are still no loops or parallel edges while others maintain that simple graphs refer only to undirected graphs. I'm assuming we use the second version of the terminology. Is this correct?

Reply:
@joshyaffee you are correct, there is also a `SimpleDiGraph` type which is the formal digraph that you are thinking of.